tmk_core/protocol/vusb/vusb.c

```diff
@@ -509,7 +524,27 @@ const PROGMEM uchar shared_hid_report[] = {
     0x95, 0x01,                //   Report Count (1)
     0x75, 0x10,                //   Report Size (16)
     0x81, 0x00,                //   Input (Data, Array, Absolute)
-    0xC0                       // End Collection
+    0xC0,                      // End Collection
+#endif
+
+#ifdef PROGRAMMABLE_BUTTON_ENABLE
+    // Programmable buttons report descriptor
+    0x05, 0x0C,                          // Usage Page (Consumer)
+    0x09, 0x01,                          // Usage (Consumer Control)
+    0xA1, 0x01,                          // Collection (Application)
+    0x85, REPORT_ID_PROGRAMMABLE_BUTTON, //   Report ID
+    0x09, 0x03,                          //   Usage (Programmable Buttons)
+    0xA1, 0x04,                          //   Collection (Named Array)
+    0x05, 0x09,                          //     Usage Page (Button)
+    0x19, 0x01,                          //     Usage Minimum (Button 1)
+    0x29, 0x20,                          //     Usage Maximum (Button 32)
+    0x14,                                //     Logical Minimum (0)
```

Comment:
The existing descriptors don't use zero length items to represent the value 0 anywhere; not sure why — maybe some implementation handles them incorrectly? Using the 1-byte form is probably safer.
```suggestion
    0x15, 0x00,                          //     Logical Minimum (0)
```